### DISCUSSION TITLE
Loading resource at run time in wasm/js targets

### DISCUSSION THREAD

**Discussion by anshul-bharat** [odin-lang/Odin#3986] [Ideas/Requests]
## Loading resource at run time in WASM/JS targets

I am working with WGPU, I am trying to develop a renderer.
 
Whenever i need to load a resource for example 3d model or texture. Currently i use #load() to load resources. It works perfectly for both native and web targets. 

But this does not seem very scalable, since it is compile time(if my understanding is correct)! It will be better, if there a function which can load resources at run time for wasm targets. 

Please if there is anything that i am not aware then please share. Currently i am new to this odin and wasm stuff.
Thanks

**Comment by laytan** [comment]
I believe I already answered your questions in discord at https://discord.com/channels/568138951836172421/1266804564557037589

We also have a HTTP package coming to core which would be usable in this case. All files are loaded over HTTP in the browser so you could have a code path that reads the assets using the `os` package for native and another one that uses the HTTP client to fetch them for the browser.
